src/UI/class-network-admin-sites-list.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Parsely Network Admin Site List class
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+use Parsely\Parsely;
+
+/**
+ * Render the additions to the WordPress Multisite Network Admin Sites List page
+ *
+ * @since 3.2.0
+ */
+final class Network_Admin_Sites_List {
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Attach network admin page functionality to the appropriate action and filter hooks.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function run() {
+		add_filter( 'manage_sites_action_links', array( __CLASS__, 'add_action_link' ), 10, 2 );
+		add_filter( 'wpmu_blogs_columns', array( __CLASS__, 'add_api_key_column' ) );
+		add_action( 'manage_sites_custom_column', array( $this, 'populate_api_key_column' ), 10, 2 );
+	}
+
+	/**
+	 * Use the manage_sites_action_links filter to append a link to the settings page in the "row actions."
+	 *
+	 * @param array $actions The list of actions meant to be displayed for the current site's context in the row actions.
+	 * @param int   $_blog_id The blog ID for the current context.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return array The list of actions including ours.
+	 */
+	public static function add_action_link( array $actions, int $_blog_id ): array {
+		if ( ! current_user_can( Parsely::CAPABILITY ) ) {
+			return $actions;
+		}
+
+		if ( ! is_numeric( $_blog_id ) ) {
+			return $actions;
+		}
```

Comment:
What do you think about merging this two conditionals into one?

Reply:
I'd go with keeping them separate. The conditions are different contexts - one is about the state of the environment (whether logged in user has certain capabilities), while the other is about the data type of a specific variable.

Reply:
I could go either way, so I'll default to leaving it as is :)